src/NUnitFramework/framework/Compatibility/ReflectionExtensions.cs

```diff
@@ -250,7 +250,9 @@ public static PropertyInfo GetProperty(this Type type, string name)
         /// <returns></returns>
         public static PropertyInfo GetProperty(this Type type, string name, BindingFlags flags)
         {
+            bool declaredOnly = flags.HasFlag(BindingFlags.DeclaredOnly);
             return type.GetRuntimeProperties()
+                .Where(prop => declaredOnly ? prop.DeclaringType.Equals(type) : true)
```

Comment:
I was considering writing this more succinctly as 
`.Where(prop => !declaredOnly || prop.DeclaringType.Equals(type))`
but chose the more verbose solution for readability (also I guess/hope that the compiler will generate the same IL).

Reply:
Good choice - I agree!

Reply:
Good choice - I agree!